testdata/src/static_msg/static_msg.go

```diff
@@ -23,13 +23,20 @@ func tests() {
 	slog.Log(ctx, slog.LevelInfo, constMsg)
 	slog.With("key", "value").Info(constMsg)
 
+	slog.Info(varMsg)                      // want `message should be a string literal or a constant`
+	slog.InfoContext(ctx, varMsg)          // want `message should be a string literal or a constant`
+	slog.Log(ctx, slog.LevelInfo, varMsg)  // want `message should be a string literal or a constant`
+	slog.With("key", "value").Info(varMsg) // want `message should be a string literal or a constant`
+
 	slog.Info(fmt.Sprintf("msg"))                      // want `message should be a string literal or a constant`
 	slog.InfoContext(ctx, fmt.Sprintf("msg"))          // want `message should be a string literal or a constant`
 	slog.Log(ctx, slog.LevelInfo, fmt.Sprintf("msg"))  // want `message should be a string literal or a constant`
 	slog.With("key", "value").Info(fmt.Sprintf("msg")) // want `message should be a string literal or a constant`
 
-	slog.Info(varMsg)                      // want `message should be a string literal or a constant`
-	slog.InfoContext(ctx, varMsg)          // want `message should be a string literal or a constant`
-	slog.Log(ctx, slog.LevelInfo, varMsg)  // want `message should be a string literal or a constant`
-	slog.With("key", "value").Info(varMsg) // want `message should be a string literal or a constant`
+	// binary expressions:
+	slog.Info("msg" + "msg")
```

Comment:
Adding check for consecutive concatenation could help 

```suggestion
	slog.Info("msg" + "msg")
	slog.Info("msg" + "msg" + "msg")
```

You added this

        slog.Info("msg" + constMsg + varMsg + fmt.Sprintf("msg")) // want `message should be a string literal or a constant`

But as an invalid one

Reply:
Agreed, having a check to exercise an extra level of recursion would be helpful.